# alhomoud_cart_rad
Scripts related to the analysis of scRNA-seq data from mice treated with CART19 cell treatment (Cy/mCART19) and with CART19 plus radiation (Cy/LD-TBI/mCART19). GFP/Luciferase transduced mouse-derived diffuse large B cell lymphoma (DLBCL) cells were transplanted to BALB/c mice and treated with either only CART19 or CART19 plus radiation with the goal of studying how radiation may interfere in the efficacy of CART19 cells in DLBCL.

![](WCM_MB_LOGO_HZSS1L_CLR_RGB_new.png)

## DATA AVAILABILITY

* Sequencing reads and processed object (raw and normalized counts) can be downloaded from **GEO: GSE291284**.

## DATA ANALYSIS

- **Step 1: QC and integration**. Removal of low quality cells and doublets, and integration using **Harmony** approach for batch correction. [QC](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/Guzman_sc_filtering_clean.ipynb)
- **Step 2: dimentionality reduction, clustering and annotation**. Dimensionality reduction, clustering and annotation steps. Annotation was run through two different approaches; 1) ORA using `PangloDB` database and 2) confirm annotations using canonical known markers for main cell types. Data that will be used in `Step 6`. [dim_red](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/Guzman_harmony_post_integration_clean.ipynb)
- **Step 3: T cells reclustering and sub-annotation**. Subset T cells after `Step 2` and find different specific T cell types (data that will be used in `Step 4` and `Step 7`). Annotation of these T cells was done using top markers and canonical markers. Then, enginereed mCART19 cells were identified by the expression of `Myc` (data that will be used in `Step 5`). Cells expressing Myc were annotated as enginereed mCART19 cells. To study if mCART19 cells were enriched in Cy/mCART19 or Cy/LD-TBI/mCART19 populations, binomial test was used followed by FDR multiple testing. [T_cells_reclust](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/tcell_reclustering_annotation_clean.ipynb)
- **Step 4: DEA for T cell clusters using pseudocount approach**. Differential Expression Analysis (DEA) was run using **DESeq2** package and using summed pseudocounts for each identified T cell subcluster between Cy/LD-TBI/mCART19 vs Cy/mCART19. [DEA_T_cells](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/pseudobulk_tcells_clean.ipynb)
- **Step 5: Global DEA for mCART19-Myc+ cells using pseudocount approach**. For myc+ cells, bulk DEA was run using summed pseudocounts between Cy/LD-TBI/mCART19 vs Cy/mCART19. [global_DEA_Myc](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/pseudobulk_CART.ipynb). [Script](ttps://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/heatmap_MYC_pos_pseudoDE_all_Tcells.R) to draw heatmap of the most meaningful genes from this analysis. 
- **Step 6: Compositional Data Analysis for main cell types**. **scCODA** approach was used to identify proportional changes in the different main cell types between Cy/LD-TBI/mCART19 and Cy/mCART19 conditons. [scCODA_main_cells](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/scCODA_compositional_data_all_cell_types.ipynb)
- **Step 7: Compositional Data Analysis for T cell clusters**. **scCODA** to identify proportional changes in T cells between Cy/LD-TBI/mCART19 and Cy/mCART19 conditons. [scCODA_T_cells](https://github.com/abcwcm/alhomoud_cart_rad/blob/main/analysis/scCODA_compositional_data_clean_t_cells.ipynb)